vllm/config.py

```diff
@@ -1721,18 +1722,13 @@ def _get_and_verify_max_len(
             default_max_len)
         derived_max_model_len = default_max_len
 
+    # Backwards compatibility
+    patch_rope_scaling(hf_config)
+
     rope_scaling = getattr(hf_config, "rope_scaling", None)
     if rope_scaling is not None:
-        if "type" in rope_scaling:
-            rope_type = rope_scaling["type"]
-        elif "rope_type" in rope_scaling:
-            rope_type = rope_scaling["rope_type"]
-        else:
-            raise ValueError(
-                "rope_scaling must have a 'type' or 'rope_type' key.")
+        rope_type = rope_scaling["rope_type"]
```

Comment:
We should add a note here that here `rope_scaling` has already been patched by `patch_rope_scaling` to assert `rope_type` is always a key of this dictionary.